Modules/@babylonjs/react-native-basekit/scripts/install.js

```diff
@@ -0,0 +1,174 @@
+    process.env.npm_config_cache,
+  ].filter(function(_) { return _; });
+}
+
+function getCached(package, binaryFilename) {
+  return [getCachePathCandidates(), 'BabylonReactNative', package.version, binaryFilename].join('/');
+}
+
+function useCachedFile(filePath) {
+  if (fs.existsSync(filePath)) {
+    console.log(`Using cached file ${filePath}`);
+    const gunzip = zlib.createGunzip();
+    const untar = tar.extract({ cwd: process.cwd() }); // Extract to the current working directory
+
+    const sourceStream = fs.createReadStream(filePath); 
+    sourceStream
+    .pipe(gunzip) // Decompress the .tar.gz file
+    .pipe(untar) // Extract the tar contents to the cwd
+    .on('error', (err) => {
+      console.error('Error extracting .tar.gz file:', err);
+      process.exit(1);
+    })
+    .on('end', () => {
+      console.log('Extraction completed successfully.');
+    });
+    return true;
+  }
+  return false;
+}
+
+function getArgument(name) {
+  const flags = process.argv.slice(2), index = flags.lastIndexOf(name);
+
+  if (index === -1 || index + 1 >= flags.length) {
+    return null;
+  }
+
+  return flags[index + 1];
+}
+
+function getBinaryUrl(package, binaryFilename) {
+  var site = getArgument('--brn-binary-site') ||
+              process.env.BRN_BINARY_SITE  ||
+              process.env.npm_config_brn_binary_site ||
+              (package.nodeBabylonConfig && package.nodeBabylonConfig.binarySite) ||
+              'https://github.com/CedricGuillemet/BabylonReactNative-1/releases/download';
+
+  return [site, package.version, binaryFilename].join('/');
+}
+
+const downloadExtractAndCache = (url, cachedFilePath) => {
+  const options = {
+    followRedirects: true, // Follow HTTP 3xx redirects
+  };
+
+  https.get(url, options, (response) => {
+    if ([301, 302, 303, 307, 308].includes(response.statusCode)) {
+      // If the response is a redirect, recursively call downloadExtractAndCache with the new URL
+      downloadExtractAndCache(response.headers.location, cachedFilePath);
+    } else if (response.statusCode === 200) {
+      const gunzip = zlib.createGunzip();
+      const untar = tar.extract({ cwd: process.cwd() }); // Extract to the current working directory
+
+      // create cache directory to store download file
+      try {
+        fs.mkdirSync(path.dirname(cachedFilePath), {recursive: true});
+      } catch (err) {
+        console.error('Unable to create directory for cache', err);
+        process.exit(1);
+      }
+
+      const chunks = [];
+      response.on('data', (chunk) =>{
+        chunks.push(chunk);
+      })
+      .on('end', (err) => {
+        const fileData = Buffer.concat(chunks);
+        fs.writeFile(cachedFilePath, fileData, (err) => {
+          if (err) {
+            console.error('Error writing BabylonReactNative cache file:', err);
+            process.exit(1);
+          }
+        });
+      })
+      .pipe(gunzip) // Unzip the response
+      .pipe(untar)  // Extract the tar file
```

Comment:
just wondering if it is the same process for windows binaries?

Reply:
yes